### PR TITLE
feat(core): Adds `getWebhooks()` function for the channel API and for the guild API

### PR DESCRIPTION
**Pull request by dager-mohamed**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**

This function requests the Discord API to fetch all webhooks associated with a specific channel. It does this by using the `id` parameter, which specifies the id of the channel for which the webhooks are to be retrieved. The `Routes.channelWebhooks(id)` method is used to construct the URL of the API endpoint that retrieves the webhooks for the specified channel. The function then makes a GET request to that URL using the `rest.get()` method, and the result of that request is returned as a promise, which will resolve to an object of type `RESTGetAPIWebhookResult`.

This request allows the retrieval of all webhooks created in a specific channel. It's possible to have multiple webhooks associated with a channel and they will be returned in an array in response to the request.

**Please move lines that apply to you out of the comment:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by Qjuh** [comment]
This should be a `getWebhooks` method on the channel API, since you don’t get a channel from a webhook but the other way around.
And while you‘re at it: a similar method is also missing on the guild API to fetch all guild webhooks.

**Comment by dager-mohamed** [comment]
> This should be a `getWebhooks` method on the channel API, since you don’t get a channel from a webhook but the other way around. And while you‘re at it: a similar method is also missing on the guild API to fetch all guild webhooks.

done added `getWebhooks()` function for the guild and channel, added for the guild to list of guild webhooks object, and for channel API to get the webhooks of the channel.

File: packages/core/src/api/channel.ts [hunk 1]
```diff
@@ -349,4 +350,14 @@ export class ChannelsAPI {
...
 	}
+
+	/**
+	 * Fetches channel webhooks
+	 *
+	 * @see {@link https://discord.com/developers/docs/resources/webhook#get-channel-webhooks}
+	 * @param id - The id of the channel
+	 */
+	public async getWebhooks(id: Snowflake) {
+		return this.rest.get(Routes.channelWebhooks(id)) as Promise<RESTGetAPIWebhookResult>;
```
Comment (almeidx) [inline packages/core/src/api/channel.ts:361]:
```suggestion
		return this.rest.get(Routes.channelWebhooks(id)) as Promise<RESTGetAPIChannelWebhooksResult>;
```

File: packages/core/src/api/channel.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import {
 	type RESTPostAPIChannelMessageCrosspostResult,
 	type RESTPostAPIChannelMessageJSONBody,
 	type RESTPostAPIChannelMessageResult,
+	type RESTGetAPIWebhookResult,
```
Comment (almeidx) [inline packages/core/src/api/channel.ts:24]:
```suggestion
	type RESTGetAPIChannelWebhooksResult,
```

File: packages/core/src/api/guild.ts [hunk 3]
```diff
@@ -83,6 +83,7 @@ import {
 	type RESTPostAPITemplateCreateGuildJSONBody,
 	type RESTPutAPIGuildBanJSONBody,
 	type RESTPutAPIGuildTemplateSyncResult,
+	type RESTGetAPIWebhookResult,
```
Comment (almeidx) [inline packages/core/src/api/guild.ts:86]:
```suggestion
	type RESTGetAPIGuildWebhooksResult,
```

File: packages/core/src/api/guild.ts [hunk 4]
```diff
@@ -965,4 +966,14 @@ export class GuildsAPI {
...
 	}
+
+	/**
+	 * Fetches guild webhooks
+	 *
+	 * @see {@link https://discord.com/developers/docs/resources/webhook#get-guild-webhooks}
+	 * @param id - The id of the guild
+	 */
+	public async getWebhooks(id: Snowflake) {
+		return this.rest.get(Routes.guildWebhooks(id)) as Promise<RESTGetAPIWebhookResult>;
```
Comment (almeidx) [inline packages/core/src/api/guild.ts:977]:
```suggestion
		return this.rest.get(Routes.guildWebhooks(id)) as Promise<RESTGetAPIGuildWebhooksResult>;
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

File: packages/core/src/api/guild.ts [hunk 5]
```diff
@@ -965,4 +966,14 @@ export class GuildsAPI {
 	public async createTemplate(templateCode: string, data: RESTPostAPITemplateCreateGuildJSONBody) {
 		return this.rest.post(Routes.template(templateCode), { body: data }) as Promise<RESTPostAPIGuildTemplatesResult>;
 	}
+
+	/**
+	 * Fetches guild webhooks
```
Comment (Jiralite) [inline packages/core/src/api/guild.ts:971]:
```suggestion
	 * Fetches webhooks for a guild
```

**Review by Jiralite — COMMENTED** [review]
Consistency with other descriptions!

**Review by suneettipirneni — APPROVED** [review]

File: packages/core/src/api/channel.ts [hunk 6]
```diff
@@ -21,6 +21,7 @@ import {
 	type RESTPostAPIChannelMessageCrosspostResult,
 	type RESTPostAPIChannelMessageJSONBody,
 	type RESTPostAPIChannelMessageResult,
+	type RESTGetAPIChannelWebhooksResult,
```
Comment (kyranet) [inline packages/core/src/api/channel.ts:24]: This import is not sorted, it should be on line 15 (`RESTGetAPIChannelUsersThreadsArchivedResult` → `RESTGetAPIChannelWebhooksResult` → `RESTPatchAPIChannelJSONBody`).

File: packages/core/src/api/guild.ts [hunk 7]
```diff
@@ -83,6 +83,7 @@ import {
 	type RESTPostAPITemplateCreateGuildJSONBody,
 	type RESTPutAPIGuildBanJSONBody,
 	type RESTPutAPIGuildTemplateSyncResult,
+	type RESTGetAPIGuildWebhooksResult,
```
Comment (kyranet) [inline packages/core/src/api/guild.ts:86]: This import isn't sorted, it should be on line 41 (`RESTGetAPIGuildMembersSearchQuery` → `RESTGetAPIGuildWebhooksResult` → `RESTGetAPIGuildWelcomeScreenResult`).

**Review by kyranet — CHANGES_REQUESTED** [review]

**Comment by dager-mohamed** [comment]
done, the imported types has been sorted
@kyranet

**Review by kyranet — APPROVED** [review]

**Comment by dager-mohamed** [comment]
@SpaceEEC
@iCrawl

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**